### PR TITLE
Merge `package:http_parser`

### PR DESCRIPTION
**Pull request by mosuem**
- [x] Move and fix workflow files, labeler.yaml, and badges in the README.md
- [x] Rev the version of the package, so that pub.dev points to the correct site
- [x] Add a line to the changelog:
```
* Move to `dart-lang/http` monorepo.
```

- [x] Add the package to the top-level readme of the monorepo:
```
| [http_parser](pkgs/http_parser/) | A platform-independent package for parsing and serializing HTTP formats. | [![pub package](https://img.shields.io/pub/v/http_parser.svg)](https://pub.dev/packages/http_parser) |
```

- [ ] **Important!** Merge the PR with 'Create a merge commit' (enabling then disabling the `Allow merge commits` admin setting)
- [ ] Update the auto-publishing settings on https://pub.dev/packages/http_parser/admin
- [ ] Add the following text to https://github.com/dart-lang/http_parser/:'

```
> [!IMPORTANT]  
> This repo has moved to https://github.com/dart-lang/http/tree/master/pkgs/http_parser
```

- [ ] Publish using the autopublish workflow
- [ ] Push tags to GitHub using `git tag --list 'http_parser*' | xargs git push origin`
- [ ] Close open PRs in dart-lang/http_parser with the following message:

```
Closing as the [dart-lang/http_parser](https://github.com/dart-lang/http_parser) repository is merged into the [dart-lang/http](https://github.com/dart-lang/http) monorepo. Please re-open this PR there!
```
      
- [ ] Transfer issues by running `dart run pkgs/repo_manage/bin/report.dart transfer-issues --source-repo dart-lang/http_parser --target-repo dart-lang/http --add-label package:http_parser --apply-changes`
- [ ] Archive https://github.com/dart-lang/http_parser/

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/http_parser.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    paths:
+      - '.github/workflows/http_parser.yaml'
+      - 'pkgs/http_parser/**'
+  schedule:
+    - cron: "0 0 * * 0"
+
+env:
+  PUB_ENVIRONMENT: bot.github
+
+permissions: read-all
```
Comment (devoncarew) [inline .github/workflows/http_parser.yaml:21]: In this file, we'll also need:

```yaml
defaults:
  run:
    working-directory: pkgs/http_parser/
```

File: .github/workflows/http_parser.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+name: package:http_parser
+
+on:
+  push:
+    branches:
+      - main
```
Comment (devoncarew) [inline .github/workflows/http_parser.yaml:6]: Why on both `master` and `main`? This repo is still using `master` (though, after all the dust has settled, we should switch it to `main`).

**Review by devoncarew — APPROVED** [review]
Generally lgtm; one question and one change requested.

File: .github/workflows/http_parser.yaml (hunk 2 above)
Comment (mosuem) [reply .github/workflows/http_parser.yaml:6]: This was purely cargo culting from the existing workflows in this repo.

**Pull request merged**